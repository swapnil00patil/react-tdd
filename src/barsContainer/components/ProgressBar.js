import React from 'react'
import styled from 'styled-components'

const Bar = styled.div`
  height: 38px;
  width: ${props => props.barValue > 100 ? 100 : props.barValue}%;
  background: ${props => props.barValue > 100 ? 'red' : '#b3d7e6'};
  margin-bottom: 10px;
  transition: width 1s ease-in-out;
`
const Wrap = styled.div`
  height: 38px;
  width: 100%;
  background: transparent;
  margin-bottom: 10px;
  border: 1px solid #000;
  position:relative;
`
const Text = styled.span`
  position:absolute;
  width: 100%;
  display: flex;
  top: 10px;
  justify-content: center;
`

class ProgressBar extends React.PureComponent {
  render () {
    const { barValue } = this.props
    return (
      <Wrap>
        <Bar barValue={barValue} />
        <Text>{barValue}%</Text>
      </Wrap>
    )
  }
}

export default ProgressBar
