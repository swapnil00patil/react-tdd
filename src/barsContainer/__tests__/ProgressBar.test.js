import React from 'react'
import { shallow, mount } from 'enzyme'

import ProgressBar from '../components/ProgressBar'

describe('ProgressBar', () => {
  it('ProgressBar should be defined', () => {
    expect(ProgressBar).toBeDefined()
  })

  it('should render exactly same number of ProgressBar', () => {
    const wrapper = mount(<ProgressBar barValue={12} />)
    expect(wrapper.find(ProgressBar).length).toBe(1)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<ProgressBar barValue={12} />)
    expect(wrapper).toMatchSnapshot()
  })
})
