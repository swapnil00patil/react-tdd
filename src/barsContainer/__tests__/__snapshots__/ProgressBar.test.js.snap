// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    barValue={12}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          barValue={12}
/>,
        <styled.span>
          12
          %
</styled.span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "barValue": 12,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            12,
            "%",
          ],
        },
        "ref": null,
        "rendered": Array [
          12,
          "%",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            barValue={12}
/>,
          <styled.span>
            12
            %
</styled.span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "barValue": 12,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              12,
              "%",
            ],
          },
          "ref": null,
          "rendered": Array [
            12,
            "%",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
